src/inmanta/data/__init__.py

```diff
@@ -4519,6 +4519,34 @@ async def get_resources_in_latest_version(
                         result.append(resource)
         return result
 
+    @classmethod
+    async def get_latest_resources_resource_type_count(cls, environment: uuid.UUID) -> JsonType:
```

Comment:
I find the name a bit confusing. It starts with `get_latest_resources`, which makes me believe that this method will return resources, but in fact it returns information about resource types.
```suggestion
    async def get_resource_type_count_for_latest_version(cls, environment: uuid.UUID) -> JsonType:
```